src/executor.ts

```diff
@@ -540,4 +544,16 @@ export class TaskExecutor {
       }
     }, this.options.startupTimeout);
   }
+
+  private async waitForAtLeastOneConfirmedProposal() {
```

Comment:
I think a more elegant solution would be to move this logic to `TaskService.run`. You have a loop there already, you can check the `marketService` in that loop for proposals, and if there are none, sleep and don't take a task from the queue.

Reply:
You are right, it will be much simpler in `TaskService`. I moved this